src/plugins/data/common/utils/abort_utils.ts

```diff
@@ -35,24 +35,12 @@ export class AbortError extends Error {
  * with any other expected errors (or completions).
  *
  * @param signal The `AbortSignal` to generate the `Promise` from
- * @param shouldReject If `false`, the promise will be resolved, otherwise it will be rejected
  */
-export function toPromise(signal: AbortSignal, shouldReject?: false): Promise<undefined | Event>;
-export function toPromise(signal: AbortSignal, shouldReject?: true): Promise<never>;
-export function toPromise(signal: AbortSignal, shouldReject: boolean = false) {
-  const promise = new Promise((resolve, reject) => {
-    const action = shouldReject ? reject : resolve;
-    if (signal.aborted) action();
-    signal.addEventListener('abort', action);
+export function toPromise(signal: AbortSignal): Promise<never> {
+  return new Promise((resolve, reject) => {
+    if (signal.aborted) reject(new AbortError());
+    signal.addEventListener('abort', () => reject(new AbortError()));
```

Comment:
I think every time we `addEventListener` we also have to have `removeEventListener` somewhere. 
Otherwise we are spamming memory leaks. 

Just double checked this is required, for example, fetch polyfill does this: https://github.com/github/fetch/blob/master/fetch.js#L534

I guess should be fixed in separate pr, since it isn't something new introduced with the refactor.

Reply:
Ahh, turns out I mentioned it here: #65051. 
Added fetch example there in description

Reply:
Added a removeEventListener

Reply:
It won't always work. We also need to remove listener after corresponding request is successfully finished. 
That is why they do it `finalise` callback. https://github.com/github/fetch/blob/master/fetch.js#L534 
Should be tacked properly in scope of https://github.com/elastic/kibana/issues/65051